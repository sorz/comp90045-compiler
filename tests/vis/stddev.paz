program stddev;

{ Compute the mean and variance of a list of numbers }

    var
        x: real;
        sum: real;
        sumsq: real;
        mean: real;

        count: integer;
        done: boolean;

begin
    write('Type a list of floats, ending with a negative number');
    writeln;

    read(x);
    count := 0;
    sum := 0;
    sumsq := 0;
    if x < 0 then
        done := true
    else
        done := false;
    while not done do
    begin
        count := count + 1;
        sum := sum + x;
        sumsq := sumsq + x*x;
        read(x);
        if x < 0 then
            done := true
        else
            done := false
    end;
    mean := sum / count;
    write('The mean is ');
    write(mean);
    writeln;
    write('The variance is ');
    write(sumsq / count - mean * mean);
    writeln
end.
