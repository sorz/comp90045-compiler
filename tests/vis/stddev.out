Type a list of floats, ending with a negative number
The mean is 2.000000
The variance is 0.666667
